Integrations developer guide (#769)
elastic/observability-docs: docs/en/integrations/what-is-an-integration.asciidoc

[[what-is-an-integration]]
= What is an integration?

> An Elastic integration is a collection of assets that defines how to observe a specific product or service with the Elastic Stack:

Reviewer: ```suggestion
An Elastic integration is a collection of assets that defines how to observe a specific product or service with the {stack}:
```

* Data ingest, storage, and transformation rules
* Configuration options
* Pre-built, custom dashboards and visualizations
* Documentation
* Tests

Integrations have a strict, well-defined structure, and offer a number of benefits over other ingest options:

* Structured around the service that is being observed--not the monitoring agent
* Easy, less error-prone configuration
* Fewer monitoring agents for users to install
* Deploy in just a few clicks

> * Decoupled release process from the Elastic Stack

Reviewer: ```suggestion
* Decoupled release process from the {stack}
```

[discrete]
[[how-integrations-work]]
== Integration lifecycle

. Create a source package
+
All integrations start as a source package.
You'll find most Elastic integrations in the {integrations-repo-link}[`elastic/integrations`] repository,
but a package can live anywhere.
+
All packages must adhere to the <<package-spec,package specification>> -- a formal spec used for the creation and validation of new or updated integrations.

. Publish the integration to the package registry
+

> Once an integration (package) has been created, it needs to be built. Built integrations are stored in the {package-storage-repo-link}[Package Storage repository] and served up via the {package-registry-repo-link}[Elastic Package Registry (EPR)].
> The Fleet UI in Kibana connects to the Package Registry and allows users to discover, install, and configure Elastic Packages.

Reviewer: ```suggestion
Once an integration (package) has been created, it needs to be built. Built integrations are stored in the {package-storage-repo-link}[Package Storage repository] and served up via the {package-registry-repo-link}[Elastic Package Registry (EPR)].
The {fleet} UI in {kib} connects to the {package-registry} and allows users to discover, install, and configure Elastic Packages.
```

The EPR can also be run locally for testing purposes.

. Install the integration
+

> Using Fleet in Kibana, install the integration and add it to an Elastic Agent policy.

Reviewer: ```suggestion
Using {fleet} in {kib}, install the integration and add it to an {agent} policy.
```

> When you install a package, its assets are unpacked and installed into Elasticsearch and Kibana using stack APIs.

Reviewer: ```suggestion
When you install a package, its assets are unpacked and installed into {es} and {kib} using {stack} APIs.
```

> In addition, configuration for the package is persisted in Elasticsearch as an Elastic Agent policy.

Reviewer: ```suggestion
In addition, configuration for the package is persisted in {es} as an {agent} policy.
```

> . Add the policy with the integration to an Elastic Agent

Reviewer: ```suggestion
. Add the policy with the integration to an {agent}.
```

+

> Once the policy with an integration is added to an Elastic Agent,

Reviewer: ```suggestion
Once the policy with an integration is added to an {agent},
```

> the agent will begin to collect and ship data to the Elastic Stack based on the Elastic integration.

Reviewer: ```suggestion
the {agent} will begin to collect and ship data to the {stack} based on the Elastic integration.
```
